app/vmui/packages/vmui/src/utils/tenants.ts

```diff
@@ -0,0 +1,9 @@
+const regexp = /(\/select\/)(\d+|\d.+)(\/)(.+)/;
+
+export const replaceTenantId = (serverUrl: string, tenantId: string) => {
+  return serverUrl.replace(regexp, `$1${tenantId}/$4`);
+};
+
+export const getTenantIdFromUrl = (url: string) => {
+  return (url.match(regexp) || [])[2];
+};
```

Comment:
I think it is not a good idea to return undefined. You can check the length of the array and return string. This string can be empty or can have a value

Reply:
Sure, good catch. I've made the correction